ext/src/ruby_api/memory.rs

```diff
@@ -131,19 +131,32 @@ impl<'a> Memory<'a> {
     /// Read +size+ bytes starting at +offset+ into a {Slice}. This
     /// provides a way to read a slice of memory without copying the data.
     ///
-    /// Resizing the memory (as with `Wasmtime::Memory#grow`) will invalidate
+    /// The returned {Slice} lazily reads the underlying memory, meaning that
+    /// the actual pointer to the string buffer is not materialzed until
+    /// `Slice#to_str` is called.
+    ///
+    /// SAFETY: Resizing the memory (as with `Wasmtime::Memory#grow`) will invalidate
     /// the slice, causing any attempt to read the slice to raise an error.
+    /// However, it is not possible to invalidate the Ruby {String} object after
```

Comment:
```suggestion
    /// However, it is not possible to invalidate the Ruby +String+ object after
```

`{}`-style is for linking to types / methods that Yard documents as part of the package -- or at least that's my understanding from using it.

Reply:
Ahhh, that makes total sense. Fixing now